license-update.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+
+# If there is already a comment block, a year range through the current year is
+# created using the first year in the comment. If there is no comment block, a
+# new one is added containing just the current year.
+
+from datetime import date
+import os
+import re
+
+sep = os.sep
+# If directory separator is backslash, escape it for regexes
+if sep == "\\":
+    sep += "\\"
+
+# Files and directories which should be included in or excluded from the update
+regexInclude = re.compile("\.cpp$|\.h$|\.hpp$|\.inc$|\.java$")
+folderExclude = "build" + sep + "|\.git" + sep + "|gradle" + sep + \
+                "|\.gradle" + sep + "|ni-libraries" + sep + "|ctre" + sep + \
+                "|frccansae" + sep + "|FRC_FPGA_ChipObject" + sep + \
+                "|gtest" + sep + "|msgs" + sep + "|i2clib" + sep + \
+                "|NetworkCommunication" + sep + "|ni" + sep + \
+                "|spilib" + sep + "|visa" + sep
+regexExclude = re.compile(folderExclude +
+                          "|NIIMAQdx\.h$|nivision\.h$|NIVisionJNI\.cpp$|"
+                          "can_proto\.h$|jni\.h$|jni_md\.h$")
+
+currentYear = str(date.today().year)
+
+# Recursively create list of files in given directory
+files = [os.path.join(dp, f) for dp, dn, fn in
+         os.walk(os.path.expanduser(".")) for f in fn]
+
+# Apply regex filters to list
+files = [f for f in files if regexInclude.search(f)]
+files = [f for f in files if not regexExclude.search(f)]
+
```

Comment:
Should we do the same thing as format.py and check for empty files list and only print "Processing" when verbose flag is provided?